tests/common.rs

```diff
@@ -538,6 +538,10 @@ pub fn compare_outputs(
             CoreTypeConcrete::Const(_) => todo!(),
             CoreTypeConcrete::BoundedInt(_) => todo!(),
             CoreTypeConcrete::Coupon(_) => todo!(),
+            CoreTypeConcrete::RangeCheck(_) => JitValue::Struct {
+                fields: vec![],
+                debug_name: None,
+            },
```

Comment:
I think you missed these ones.

Reply:
I think this needs to remain no? either keep it OR return an empty struct if the `vm_result` ty is a builtin

Reply:
If you follow the usage of that function you end up at `compare_inputless_program` which runs the program in both the VM and Cairo native then compares the results.

Both the VM and native handle builtins internally, which means that the builtins should never be accessible from outside. That's why I'm pretty sure that it's not necessary because all of those are unreachable too at that point.

You can probably test it by running a test that has the `RangeCheck` builtin.

Reply:
I did, the issue is actually the test that was uncommented in the PR, `tests/cases/cairo_vm/primitive_types2.cairo`. If you don't add this or handle the builtin ty in the compare_outputs function, you will end up hitting the `todo!`:
- the return type of the test is a range check builtin, which leads [this](https://github.com/lambdaclass/cairo_native/blob/main/tests/common.rs#L548) to be `RangeCheck`.
- the function is a `Success` and doesn't panic, so you end up calling `map_vm_values` but with `ty = RangeCheck` and `values = []` (because no return value from the vm).
- This makes you hit the `todo!()`.

Maybe it makes more since to skip the `map_vm_values` call if the return ty is a builtin?

Reply:
Okay I think I understand the issue here. We're passing the type we want to "deserialize" to `map_vm_values`, which afaik it's not supposed to contain the builtins. This means that the fact that `map_vm_values` receives a builtin type is a bug.

To avoid this we can change the conditions necessary to call `map_vm_values`. For example, here we have the deserialization of VM values for non-panic results. Can you try to change it so that it checks for builtins? Something like this should work:
```rs
if returns_panic {
    // ...
} else if !is_builtin {
    // ...
} else {
    // Generate a hardcoded unit value without calling `map_vm_values`.
}
```